Check size in bytes as opposed to string size

aardvarkk opened the pull request:
`size` returns length in characters, but doesn't factor in multibyte Unicode characters.
By switching to `bytesize`, we check the relevant measure of how many bytes the worksheet name is.

Fixes #588.

marshall-lee committed:
Fix Relationship.instances cache.

This PR aims to fix several issues with Relationship cache:

1) It's not threadsafe, so I propose to use a TLS variable for this.
2) Memory obtained by cache remains non-freed before the next run of `serialize`. I think it should be freed immediately.
3) Memory should be freed in `ensure` block to prevent memory bloating in case of exception.

*There are only two hard things in Computer Science: cache invalidation and naming things.*

marshall-lee committed:
Cache relationship in Hash rather than in Array.

Also cacle only ids, not entire instances.

straydogstudio committed:
Add community caxlsx notice

straydogstudio committed:
Update README.md

straydogstudio committed:
Update README.md

straydogstudio committed:
Release 3.0.0

straydogstudio committed:
Merge branch 'master' of https://github.com/caxlsx/axlsx

straydogstudio committed:
Change gem name

straydogstudio committed:
Update README.md

inopinatus committed:
Bump dependencies for nokogiri, rubyzip, and Ruby.

nokogiri <1.10.4 and rubyzip < 1.3.0 had CVEs and/or regressions,
and whilst we're here, signal that it's okay to use Ruby 2.3 syntax
and methods and later.

simi committed:
Update README.md to show Travis CI status of this fork.

kiskoza committed:
Add caxlsx.rb to make auto require work

https://guides.rubygems.org/make-your-own-gem/
The convention is to have one Ruby file with the same name as your gem, since that gets loaded when require is run.

takkanm committed:
add and update some ruby versions

takkanm committed:
add some ruby versions

straydogstudio committed:
Gemspec home page fix [skip ci]

pwim committed:
Fix URL in README (#6)

noniq committed:
Update gem install instructions to use caxlsx

straydogstudio committed:
Merge pull request #3 from kiskoza/fix-rails-auto-require

Add caxlsx.rb to make auto require work

straydogstudio committed:
Fix github links / repo names [skip ci]

straydogstudio committed:
Merge pull request #2 from inopinatus/master

Bump dependencies for nokogiri, ruby, and rubyzip.

straydogstudio committed:
Merge pull request #10 from marshall-lee/caxlsx-fix_relationship_cache

Fix Relationship.instances cache

straydogstudio committed:
Remove ruby 2.1 and 2.2 from Travis

straydogstudio committed:
Version 3.0.1 bump [skip ci]

straydogstudio committed:
Readme title fixes [skip ci]

straydogstudio committed:
Fix rake reference [skip ci]

alimujtaba committed:
fix validate document with font option underline

vzvu3k6k committed:
Remove a typo from comment

Fixnum is removed by 12d6433, but this typo was remained.

westonganger committed:
Use CHANGELOG.md instead of README for changelog

westonganger committed:
Improve README

westonganger committed:
Remove rubima.md

westonganger committed:
Add Style Reference

westonganger committed:
Delete appveyor.yml

noniq committed:
Create CONTRIBUTING.md

noniq committed:
Add link to contribution guidelines

noniq committed:
Add link to FAQ

noniq committed:
Add section about where to (not) post questions and find answers.

noniq committed:
Remove unused / stubbed code related to parsing xlsx files

This code was added in 3def8f8895 (back in 2011). It has never been worked on further since; `Axlsx::Parser` is not used anywhere in the whole codebase.

Currently there are no plans to support parsing xlsx file anytime soon, so let‘s remove all of this dead code.

noniq committed:
Add GitHub action for closing stale issues

noniq committed:
Add changelog entry for 2.0.2

noniq committed:
Mention axslx_styler in related gems section

noniq committed:
Fix typo.

See https://github.com/randym/axlsx/pull/617

noniq committed:
Fix typo.

See https://github.com/randym/axlsx/pull/635

noniq committed:
Fix typo

See https://github.com/randym/axlsx/pull/606

noniq commented:
@aardvarkk Would you like to recreate this PR in https://github.com/caxlsx/caxlsx/ ? I think we should get this merged.

noniq committed:
Fix rdoc for BarSeries#shape

Commit:
Check size in bytes as opposed to string size

- `size` returns length in characters, but doesn't factor in multibyte Unicode characters.
By switching to `bytesize`, we check the relevant measure of how many bytes the worksheet name is.
- Fixes https://github.com/randym/axlsx/issues/588
- Copy of PR against original axlsx
(https://github.com/randym/axlsx/pull/589)

aardvarkk commented:
@noniq Done! Please see https://github.com/caxlsx/caxlsx/pull/35

Commit:
Add test for name too long with Unicode characters
